### PR TITLE
Svelte: implement symbol redesign

### PR DESCRIPTION
**Pull request by camdencheek**
This implements the designs for the new symbol kind icons [here](https://www.figma.com/file/EPgl0qVbqGSumvf52brhda/Code-Search?type=design&node-id=3027%3A85653&mode=design&t=hKVca2TmuCChXvqd-1).

screenshot-2024-05-07_16-51-12@2x
screenshot-2024-05-07_16-50-41@2x
screenshot-2024-05-07_17-03-31@2x
screenshot-2024-05-07_17-04-00@2x

Closes https://github.com/sourcegraph/sourcegraph/issues/61941

## Test plan

Added storybook test. Manually tested the two places this was slotted in.

### CODE REVIEW

**Comment by taiyab** [comment]
I checked out the branch here but didn't see any of the new symbol icons in search results, etc.

The icons visually look like they use actual text over being true outlined svgs? Apologies if this is wrong, I can't clearly see where the source for each individual icon is.

**Comment by taiyab** [comment]
> Did you want us to just copy the paths straight from the designs? I was thinking we wanted to parameterize these so we could easily adjust the short names, colors, spacing, etc. Totally fine if that's what you want, just misunderstood 🙂

As long as it's strictly not adopting any styling from the stylesheets, then fine. To my eye I'm noticing some font-weight weirdness/faux bolding but could be wrong.

**Review by taiyab — COMMENTED** [review]
LGTM!

**Comment by camdencheek** [comment]
Converting to draft since we decided to just use the SVGs from Figma for now. We can't guarantee that our code font exists (it didn't on my system 🙃), so until we get fonts figured out, we're just going to use what we know looks good, which are the paths from Figma

**Comment by camdencheek** [comment]
Alrighty, this is ready for review again

**Review by taiyab — APPROVED** [review]
**Much** better.

**Comment by taiyab** [comment]
Excited about getting this in! :D

File: client/web-sveltekit/src/lib/search/SymbolKindIcon.svelte [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+</Tooltip>
+
+<style lang="scss">
+    div {
+        display: contents;
+
+        // TODO(@taiyab): incorporate these colors into the semantic colors
+
+        :global(.theme-light) & {
+            --icon-color: var(--text-muted);
```
Comment (vovakulikov) [inline client/web-sveltekit/src/lib/search/SymbolKindIcon.svelte:117]: Can we move --icon-color: var(--text-muted) on the top level, it seems we set it 
both times for dark and light theme

**Review by vovakulikov — APPROVED** [review]

File: client/web-sveltekit/src/lib/search/SymbolKindIcon.svelte (hunk 1 above)
Comment (camdencheek) [reply client/web-sveltekit/src/lib/search/SymbolKindIcon.svelte:117]: Done!

**Comment by taiyab** [comment]
Quick note on this @camdencheek: The symbols weren't updated in places like the search suggestions:
CleanShot 2024-05-09 at 19 48 54@2x

There might've been other places missed out.

**Comment by camdencheek** [comment]
Oh weird. Those must use a different source for the icons because I deleted the code that should be handling that 😄

**Pull request merged**